ansible/roles/python/tasks/main.yml

```diff
@@ -23,6 +23,33 @@
   - '{{ ansible_os_family|lower }}-{{ ansible_distribution_major_version }}.yml'
   - '{{ ansible_os_family|lower }}.yml'
 
+- name: Install Python
+  ansible.builtin.shell: |
+    #!/bin/bash
+
+    INSTALL_DIR="/slurm/python"
+    mkdir -p "${INSTALL_DIR}"
+    cd "${INSTALL_DIR}" || exit 1
+
+    curl -LO "https://www.python.org/ftp/python/3.13.1/Python-3.13.1.tgz"
+    tar -xzf "Python-3.13.1.tgz" -C "${INSTALL_DIR}"
+    cd "$INSTALL_DIR/Python-3.13.1" || exit 1
+    ./configure --prefix="/slurm/python"
+    make altinstall
+    cd "${INSTALL_DIR}" || exit 1
+    rm -rf Python-3.13.1*
+  args:
+    executable: /bin/bash
+
+- name: Create Python venv
```

Comment:
You should be able to eliminate this explicit virtual environment creation in favor of just setting pip on a venv that doesn't yet exist. I suggest using `--copies` arg.

```
  - name: Create virtualenv for tool
    ansible.builtin.pip:
      name: pip==X.Y.Z
      virtualenv: "{{ virtualenv_path }}"
      virtualenv_command: /slurm/python/bin/python3.13 -m venv --copies
```

https://docs.ansible.com/ansible/latest/collections/ansible/builtin/pip_module.html#parameter-virtualenv

Reply:
For clarity, I also suggest a fixed version of pip (or something like ~X.Y that will upgrade patch releases) because the pip resolver has changed over the years and it will reduce surprises.

Reply:
Sounds good, I'm also going to forgo the "Upgrade Pip" for this very reason so we can specify a specific pip version for installation.